### PR TITLE
Fixing Issue 483 For Defining HeavyWeight Test Groups

### PR DESCRIPTION
**Pull request by nairashwin952013**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,10 +2,40 @@
...
+ - Updated Prune-Depth Configuration Documentation.Also, added unit tests for additional limit values.
 ### Fixed
- - [Content-Type is not set to application/octet-stream when doing CSE](https://github.com/joyent/java-manta/issues/486) 
-
+ - [Content-Type is not set to application/octet-stream when doing CSE](https://github.com/joyent/java-manta/issues/486)
+ - [Test Groups that include Heavyweight Tests that are not run by default](https://github.com/joyent/java-manta/issues/483) 
+ - [Invalid MantaUser results in 500 Internal Server Error](https://github.com/joyent/java-manta/issues/123)
+ - [Change all hard coded "/" values to MantaClient.SEPARATOR ](https://github.com/joyent/java-manta/issues/363)  
+### Changed
+  - [Upgraded upstream dependencies to latest stable versions:](https://github.com/joyent/java-manta/pull/485)
```
Comment (dekobon) [inline CHANGELOG.md]: Please don't list the dependencies. Just link to the commit that lists them.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,40 @@
...
 
-## [3.3.2] -
+## [3.4.0] -
+### Added
+ - Updated Prune-Depth Configuration Documentation.Also, added unit tests for additional limit values.
 ### Fixed
- - [Content-Type is not set to application/octet-stream when doing CSE](https://github.com/joyent/java-manta/issues/486) 
-
+ - [Content-Type is not set to application/octet-stream when doing CSE](https://github.com/joyent/java-manta/issues/486)
+ - [Test Groups that include Heavyweight Tests that are not run by default](https://github.com/joyent/java-manta/issues/483)
```
Comment (dekobon) [inline CHANGELOG.md]: #483 Isn't fixed in this release, right? Let's omit it from the changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,40 @@
 All notable changes to this project will be documented in this file.
 This project aims to adhere to [Semantic Versioning](http://semver.org/).
 
-## [3.3.2] -
+## [3.4.0] -
+### Added
+ - Updated Prune-Depth Configuration Documentation.Also, added unit tests for additional limit values.
```
Comment (dekobon) [inline CHANGELOG.md]: Missing a space after the period.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientFindIT.java [hunk 4]
```diff
@@ -32,7 +32,7 @@
 
 import static com.joyent.manta.client.MantaClient.SEPARATOR;
 
-@Test
+@Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientFindIT.java]: Are you sure we want the default setting for all new tests in this class to be "nightly"?

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java [hunk 5]
```diff
@@ -70,7 +70,7 @@ public void afterClass() throws IOException {
         IntegrationTestConfigContext.cleanupTestDirectory(mantaClient, testPathPrefix);
     }
 
-    @Test
+    @Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java [hunk 6]
```diff
@@ -104,7 +104,7 @@ public final void testPutWithErrorProneCharacters() throws IOException {
         }
     }
 
-    @Test
+    @Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java [hunk 7]
```diff
@@ -271,7 +271,7 @@ public final void testPutWithByteArrayAndContentType() throws IOException {
...
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java [hunk 8]
```diff
@@ -289,7 +289,7 @@ public final void testPutWithPlainTextFileUTF8RomanCharacters() throws IOExcepti
                 "Uploaded file didn't match expectation");
     }
 
-    @Test
+    @Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientPutIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientRangeIT.java [hunk 9]
```diff
@@ -177,7 +177,7 @@ public final void canGetWithZeroRange() throws IOException {
         }
     }
 
-    @Test
+    @Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientRangeIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaObjectOutputStreamIT.java [hunk 10]
```diff
@@ -29,7 +29,7 @@
 import java.security.MessageDigest;
 import java.util.UUID;
 
-@Test
+@Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaObjectOutputStreamIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/multipart/ServerSideMultipartManagerIT.java [hunk 11]
```diff
@@ -211,6 +212,7 @@ public final void canListParts() throws IOException {
         }
     }
 
+    @Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/multipart/ServerSideMultipartManagerIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/multipart/ServerSideMultipartManagerIT.java [hunk 12]
```diff
@@ -281,6 +283,7 @@ public final void canUploadWithByteArrayAndMultipleParts() throws IOException {
...
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/multipart/ServerSideMultipartManagerIT.java]: This shouldn't be nightly.

File: java-manta-it/src/test/java/com/joyent/manta/client/multipart/ServerSideMultipartManagerIT.java [hunk 13]
```diff
@@ -307,6 +310,7 @@ public void errorWhenMissingPart() throws IOException {
         assertTrue(thrown, "Exception wasn't thrown");
     }
 
+    @Test(groups = { "nightly" })
```
Comment (dekobon) [inline java-manta-it/src/test/java/com/joyent/manta/client/multipart/ServerSideMultipartManagerIT.java]: This shouldn't be nightly.

**Comment by dekobon** [comment]
My initial list of nightly tests would be:
```
MantaClientJobIT.canListAllJobIDs
MantaClientJobIT.canListAllJobs
MantaClientJobIT.canListAllRunningJobIDs
MantaClientJobIT.canListAllRunningJobs
EncryptedJobsMultipartManagerIT.*
JobsMultipartManagerIT.*
ServerSideMultiparManagerIT.canListUploadsInProgress
```

**Comment by nairashwin952013** [comment]
This Pull- Request will be followed up by #504 .

**Pull request closed without merging**